pkg/slareport/s3.go

```diff
@@ -0,0 +1,70 @@
+package slareport
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"net/url"
+	"time"
+
+	"github.com/minio/minio-go/v7"
+	"github.com/minio/minio-go/v7/pkg/credentials"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+// PDFUploader uploads pdfs to the configured S3 endpoint.
+type PDFUploader struct {
+	client *minio.Client
+	bucket string
+	ctx    context.Context
+}
+
+// PDF is contains all metainformation to upload a PDF to S3.
+type PDF struct {
+	Customer string
+	Date     time.Time
+	PDFData  io.ReadCloser
+}
+
+// Login initializes an S3 client.
+func (p *PDFUploader) Login(ctx context.Context, endpoint, bucket, keyID, secretKey string) error {
```

Comment:
Nitpick: I'd probably make this a constructor function instead of a method, the same way the minio library does it. It's IMO a bit awkward to call `Login` on the Uploader.